querybook/webapp/components/TableUploader/TableUploaderConfirmForm.tsx

```diff
@@ -117,8 +120,99 @@ export const TableUploaderConfirmForm: React.FC = () => {
                     Confirm Upload Details
                 </StyledText>
             </div>
+            <TableUploaderConfirmWarnings formValues={values} />
             {renderSourceInfoDOM()}
             {renderTableSpecDOM()}
         </div>
     );
 };
+
+const TableUploaderConfirmWarnings: React.FC<{
+    formValues: ITableUploadFormikForm;
+}> = ({ formValues }) => {
+    const {
+        metastore_id: metastoreId,
+        table_config: {
+            schema_name: schemaName,
+            table_name: tableName,
+            if_exists: ifExists,
+        },
+    } = formValues;
+
+    const { data: schemaAndTableExists, isLoading } = useResource(
+        useCallback(
+            () =>
+                TableResource.checkIfExists(metastoreId, schemaName, tableName),
+            [metastoreId, schemaName, tableName]
+        )
+    );
+
+    if (isLoading) {
+        return null;
+    }
+
+    const [schemaExists, tableExists] = schemaAndTableExists;
+    let warningDOM: React.ReactNode = null;
+
+    if (!schemaExists) {
+        warningDOM = (
+            <Message
+                type="error"
+                title={
+                    <span className="flex-row">
+                        <Icon name="AlertCircle" size={20} className="mr8" />
+                        Failed to find schema {schemaName} in metastore
+                    </span>
+                }
+            >
+                The create table will fail if the schema does not exist. Please
+                run <i>create database {schemaName}</i> to create it (or any
+                applicable query for the DB you are using).
+            </Message>
+        );
+    } else if (tableExists) {
+        if (ifExists === 'fail') {
+            warningDOM = (
+                <Message
+                    type="error"
+                    title={
+                        <span className="flex-row">
+                            <Icon
+                                name="AlertCircle"
+                                size={20}
+                                className="mr8"
+                            />
+                            The table {schemaName}.{tableName} was detected in
+                            metastore
+                        </span>
+                    }
+                >
+                    Since the if exists policy is "fail". The table upload may
+                    not work.
```

Comment:
since the user already selected 'fail' for the if exists policy, why would we like to show a warning instead of just failing it so they can not proceed?

Reply:
hmm, i guess we could, i made it soft since the check is not in real time